### DISCUSSION TITLE
need some more clarity for using linkdir and linkdata options to search for torrents based on downloads

### DISCUSSION THREAD

**Discussion by goffy59** [cross-seed/cross-seed#420] [Q&A] [answered]
I am simply not sure how to use this feature and there website doesn't really provide any good references or examples as to how this is to be used. If I can't use it in daemon mode, how do I run it and what directories for example do I need to add to my container and then what setting do i use in the config for that said directory?  I have a lot more questions than answers and if i search here on github or google I dont find much in regards to this new feature. The new feature seems amazing but i just have no idea how to really use it. Does anyone have any examples such as their config or their docker run command /etc?

**Comment by swannie-eire** [comment]
I am also interested to know this.

**Comment by zakkarry** [comment, marked as answer]
https://www.cross-seed.org/docs/tutorials/data-based-matching

https://github.com/cross-seed/cross-seed/discussions/465#discussioncomment-6868960 - good writeup by another user too


### deluge bash script 
- execute plugin with path to bash script as script to run "/path/to/script.sh'
- will need to restart deluge after enabling execute "complete" script

#!/bin/bash
torrentid=$1
torrentname=$2
torrentpath=$3
curl -XPOST http://localhost:2468/api/webhook --data-urlencode "path=$torrentpath/$torrentname"

### qbittorrent
- add this to on execute in options

curl -XPOST http://localhost:2468/api/webhook --data-urlencode "path=%F"

### config
- matchMode is "risky" or "safe" - depending on your indexers consistency
- linkType is dependant on your setup functioning with the type of linking

matchMode: "risky",
    skipRecheck: false,
    linkType: "hardlink",
    linkDir: "/media/+CROSS+",
    dataDirs: [
    "/media/+UNSORTED+"
    ],

**Comment by goffy59** [comment]
Well unfortunately with the changes to Qbittorrent and no I wont go back (sqlite is amazingly better), it still wont work. I Havent been using cross-seed for months due to this. Would love to try it again but i wont use another torrent client and I wont turn off sqlite. :( Thanks for your efforts though, it may just help someone else.

**Reply by zakkarry** [reply]
I wasn't proposing that you switch clients, I offered you both a qbittorrent and deluge method for doing this.

Did you miss the second heading labeled "qbittorrent" ?

**Reply by goffy59** [reply]
Awhile back before I submitted this Qbittorrent switched to SQLITE for its torrent DB which has effected cross-seeds way of interacting with the program. I still find your answer totally legit and valuable but Qbittorrent changed some things. PLease correct me if I am wrong. :(

**Reply by zakkarry** [reply]
I'm not running the latest version of qbittorrent, so I'm not familiar.

Alternatively, you could save a copy of all torrents to a directory in qbittorrent, and use that as your `torrentDir` - although I'm unsure what you would do about the current torrents. 

There are scripts and methods to export your currently seeded torrents, however none are one-click as far as I'm aware.
